### ISSUE TITLE
Errors which prevent installing Pro Apps in some conditions reproduced on XWiki 16.4.0

### ISSUE THREAD

**Issue by ilieandriuta** [xwikisas/application-licensing#175]
Environment: Windows 11 Pro, Chrome 126, using a local instance of XWiki 16.4.0 on PostgreSQL 16, Tomcat 9.0.89, Java 21 (Oracle) , Licensor Application 1.25

STEPS TO REPRODUCE

1. Start an instance of XWiki 16.4.0 (tested with PostgreSQL 16/ Tomcat 9.0.89)
2. Install Active Directory Application (tested with 1.16.14) or other Pro App (reproduced also with Diagram Pro)
3. Observe the Extension Manager logs and XWiki console

EXPECTED RESULTS

No errors/ warnings are displayed.

ACTUAL RESULTS

Sometimes, in Extension Manager logs there are some warnings and errors that prevent the installation of the Pro App (and others, tried also with Diagram):

![image](https://github.com/xwikisas/application-licensing/assets/32512703/ed67d161-73de-40dd-9a89-6b05f2f4db3c)

Full stacktrace from EM logs:

```
Exception thrown during job execution
class org.xwiki.extension.InstallException: Failed to resolve extension
    at org.xwiki.extension.job.internal.AbstractInstallPlanJob.installExtension(AbstractInstallPlanJob.java:913)
[...]
    at java.base/java.lang.Thread.run(Thread.java:1583)
Caused by: class org.xwiki.extension.InstallException: Failed to create an install plan for extension dependency [com.xwiki.activedirectory:application-activedirectory-api-1.16.14]
    at org.xwiki.extension.job.internal.AbstractInstallPlanJob.installExtensionDependency(AbstractInstallPlanJob.java:805)
   [...]
    at java.base/java.lang.Thread.run(Thread.java:1583)
Caused by: class org.xwiki.extension.InstallException: Failed to create an install plan for extension dependency [com.xwiki.licensing:application-licensing-licensor-api-1.25]
    at org.xwiki.extension.job.internal.AbstractInstallPlanJob.installExtensionDependency(AbstractInstallPlanJob.java:805)
    [...]
    at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
    at java.base/java.lang.Thread.run(Thread.java:1583)
Caused by: class org.xwiki.extension.InstallException: Failed to create an install plan for extension dependency [com.xwiki.licensing:application-licensing-licensor-ui-1.25]
    at org.xwiki.extension.job.internal.AbstractInstallPlanJob.installExtensionDependency(AbstractInstallPlanJob.java:805)
    [...]
    at java.base/java.lang.Thread.run(Thread.java:1583)
Caused by: class org.xwiki.extension.InstallException: Failed to create an install plan for extension dependency [org.xwiki.platform:xwiki-platform-icon-fontawesome-16.4.0]
    at org.xwiki.extension.job.internal.AbstractInstallPlanJob.installExtensionDependency(AbstractInstallPlanJob.java:805)
   [...]
    at java.base/java.lang.Thread.run(Thread.java:1583)
Caused by: class java.lang.IllegalStateException: null
    at java.base/java.util.LinkedHashMap$LinkedHashIterator.remove(LinkedHashMap.java:1034)
    at org.xwiki.extension.job.internal.AbstractInstallPlanJob.installExtension(AbstractInstallPlanJob.java:1095)
   [...]
    at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
    at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
    at java.base/java.lang.Thread.run(Thread.java:1583)
```
Full stacktrace available at: https://up1.xwikisas.com/#aQ0BLuAFen6M0GcdQbnFDw

The error doesn't reproduce every time and I didn't find a pattern to reproduce it regularly (for ex. I reproduced it twice yesterday in different moments of the day with the same stacktrace, but I don't know  if it's related to servers being down or not). 

Also, it did not seem to reproduce on XWiki 16.4.0 Jetty/HSQL demo instances or 15.10.11 demo instances, tried also on a 16.4.0 Jetty/MariaDB.

@ane-gabriela reproduced it as well on a 16.4.0 Jetty/MySQL 8 instance, **along with the following errors** in XWiki console which **I personally reproduced when uninstalled Google Apps Integration Pro** (but I don't know if it's related to this issue).

```
2024-07-02 16:47:06,254 [http-nio-1115-exec-7 - http://localhost:1115/xwiki/rest/liveData/sources/liveTable/entries?timestamp=1719928025842&namespace=wiki%3Axwiki&sourceParams.resultPage=XWiki.XWikiUserMembershipSheet&sourceParams.%24doc=XWiki.ILIE&sourceParams.translationPrefix=user.profile.groups.table.&sourceParams.queryFilters=currentlanguage%2Chidden&properties=group&offset=0&limit=15&sort=group&descending=] ERROR o.h.h.i.a.ErrorTracker         -  Invalid path: 'obj.id'
2024-07-02 16:47:06,255 [http-nio-1115-exec-7 - http://localhost:1115/xwiki/rest/liveData/sources/liveTable/entries?timestamp=1719928025842&namespace=wiki%3Axwiki&sourceParams.resultPage=XWiki.XWikiUserMembershipSheet&sourceParams.%24doc=XWiki.ILIE&sourceParams.translationPrefix=user.profile.groups.table.&sourceParams.queryFilters=currentlanguage%2Chidden&properties=group&offset=0&limit=15&sort=group&descending=] ERROR o.h.h.i.a.ErrorTracker         -  Invalid path: 'obj.id'
org.hibernate.hql.internal.ast.InvalidPathException: Invalid path: 'obj.id'
        at org.hibernate.hql.internal.ast.util.LiteralProcessor.lookupConstant(LiteralProcessor.java:111)
        at org.hibernate.hql.internal.ast.tree.DotNode.resolve(DotNode.java:218)
        at org.hibernate.hql.internal.ast.HqlSqlWalker.resolve(HqlSqlWalker.java:1066)
      [...]
        at org.apache.tomcat.util.threads.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1190)
        at org.apache.tomcat.util.threads.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:659)
        at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:63)
        at java.base/java.lang.Thread.run(Thread.java:1583)
2024-07-02 16:47:06,278 [http-nio-1115-exec-7 - http://localhost:1115/xwiki/rest/liveData/sources/liveTable/entries?timestamp=1719928025842&namespace=wiki%3Axwiki&sourceParams.resultPage=XWiki.XWikiUserMembershipSheet&sourceParams.%24doc=XWiki.ILIE&sourceParams.translationPrefix=user.profile.groups.table.&sourceParams.queryFilters=currentlanguage%2Chidden&properties=group&offset=0&limit=15&sort=group&descending=] ERROR o.h.h.i.a.ErrorTracker         - <AST>:0:0: unexpected end of subtree
2024-07-02 16:47:06,299 [http-nio-1115-exec-7 - http://localhost:1115/xwiki/rest/liveData/sources/liveTable/entries?timestamp=1719928025842&namespace=wiki%3Axwiki&sourceParams.resultPage=XWiki.XWikiUserMembershipSheet&sourceParams.%24doc=XWiki.ILIE&sourceParams.translationPrefix=user.profile.groups.table.&sourceParams.queryFilters=currentlanguage%2Chidden&properties=group&offset=0&limit=15&sort=group&descending=] ERROR o.h.h.i.a.ErrorTracker         - <AST>:0:0: unexpected end of subtree
antlr.NoViableAltException: unexpected end of subtree
        at org.hibernate.hql.internal.antlr.HqlSqlBaseWalker.exprOrSubquery(HqlSqlBaseWalker.java:4879)
        at org.hibernate.hql.internal.antlr.HqlSqlBaseWalker.comparisonExpr(HqlSqlBaseWalker.java:4264)
        at org.hibernate.hql.internal.antlr.HqlSqlBaseWalker.logicalExpr(HqlSqlBaseWalker.java:2180)
     [...]
        at org.xwiki.rendering.internal.transformation.macro.MacroTransformation.transform(MacroTransformation.java:311)
        at org.xwiki.rendering.internal.transformation.DefaultRenderingContext.transformInContext(DefaultRenderingContext.java:183)
        at org.xwiki.rendering.internal.transformation.DefaultTransformationManager.performTransformations(DefaultTransformationManager.java:88)
        at org.xwiki.display.internal.DocumentContentAsyncExecutor.executeInCurrentExecutionContext(DocumentContentAsyncExecutor.java:396)
        at org.xwiki.display.internal.DocumentContentAsyncExecutor.execute(DocumentContentAsyncExecutor.java:269)
        at org.xwiki.display.internal.DocumentContentAsyncRenderer.execute(DocumentContentAsyncRenderer.java:112)
        at org.xwiki.rendering.async.internal.block.AbstractBlockAsyncRenderer.render(AbstractBlockAsyncRenderer.java:157)
        at org.xwiki.rendering.async.internal.block.AbstractBlockAsyncRenderer.render(AbstractBlockAsyncRenderer.java:54)
        at org.xwiki.rendering.async.internal.DefaultAsyncRendererExecutor.syncRender(DefaultAsyncRendererExecutor.java:290)
        at org.xwiki.rendering.async.internal.DefaultAsyncRendererExecutor.render(DefaultAsyncRendererExecutor.java:267)
        at org.xwiki.rendering.async.internal.block.DefaultBlockAsyncRendererExecutor.execute(DefaultBlockAsyncRendererExecutor.java:125)
        at org.xwiki.display.internal.DocumentContentDisplayer.display(DocumentContentDisplayer.java:67)
        at org.xwiki.display.internal.DocumentContentDisplayer.display(DocumentContentDisplayer.java:43)
        at org.xwiki.display.internal.DefaultDocumentDisplayer.display(DefaultDocumentDisplayer.java:96)
        at org.xwiki.display.internal.DefaultDocumentDisplayer.display(DefaultDocumentDisplayer.java:39)
        at org.xwiki.sheet.internal.SheetDocumentDisplayer.display(SheetDocumentDisplayer.java:123)
        at org.xwiki.sheet.internal.SheetDocumentDisplayer.display(SheetDocumentDisplayer.java:52)
        at org.xwiki.display.internal.ConfiguredDocumentDisplayer.display(ConfiguredDocumentDisplayer.java:68)
        at org.xwiki.display.internal.ConfiguredDocumentDisplayer.display(ConfiguredDocumentDisplayer.java:42)
        at com.xpn.xwiki.doc.XWikiDocument.display(XWikiDocument.java:1388)
        at com.xpn.xwiki.doc.XWikiDocument.getRenderedContent(XWikiDocument.java:1524)
        at com.xpn.xwiki.doc.XWikiDocument.getRenderedContent(XWikiDocument.java:1491)
        at com.xpn.xwiki.doc.XWikiDocument.getRenderedContent(XWikiDocument.java:1406)
        at org.xwiki.livedata.internal.livetable.LiveTableLiveDataResultsRenderer.lambda$getLiveTableResultsFromPage$0(LiveTableLiveDataResultsRenderer.java:107)
        at org.xwiki.livedata.internal.livetable.LiveTableRequestHandler.getLiveTableResults(LiveTableRequestHandler.java:114)
        at org.xwiki.livedata.internal.livetable.LiveTableLiveDataResultsRenderer.getLiveTableResultsFromPage(LiveTableLiveDataResultsRenderer.java:89)
        at org.xwiki.livedata.internal.livetable.LiveTableLiveDataEntryStore.getLiveTableResultsJSON(LiveTableLiveDataEntryStore.java:139)
        at org.xwiki.livedata.internal.livetable.LiveTableLiveDataEntryStore.get(LiveTableLiveDataEntryStore.java:106)
        at org.xwiki.livedata.internal.rest.DefaultLiveDataEntriesResource.getEntries(DefaultLiveDataEntriesResource.java:200)
        at org.xwiki.livedata.internal.rest.DefaultLiveDataEntriesResource.getEntries(DefaultLiveDataEntriesResource.java:79)
        at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
        at java.base/java.lang.reflect.Method.invoke(Method.java:580)
        at org.glassfish.jersey.server.model.internal.ResourceMethodInvocationHandlerFactory.lambda$static$0(ResourceMethodInvocationHandlerFactory.java:52)
        at org.glassfish.jersey.server.model.internal.AbstractJavaResourceMethodDispatcher$1.run(AbstractJavaResourceMethodDispatcher.java:146)
        at org.glassfish.jersey.server.model.internal.AbstractJavaResourceMethodDispatcher.invoke(AbstractJavaResourceMethodDispatcher.java:189)
        at org.glassfish.jersey.server.model.internal.JavaResourceMethodDispatcherProvider$TypeOutInvoker.doDispatch(JavaResourceMethodDispatcherProvider.java:219)
        at org.glassfish.jersey.server.model.internal.AbstractJavaResourceMethodDispatcher.dispatch(AbstractJavaResourceMethodDispatcher.java:93)
        at org.glassfish.jersey.server.model.ResourceMethodInvoker.invoke(ResourceMethodInvoker.java:478)
        at org.glassfish.jersey.server.model.ResourceMethodInvoker.apply(ResourceMethodInvoker.java:400)
        at org.glassfish.jersey.server.model.ResourceMethodInvoker.apply(ResourceMethodInvoker.java:81)
       [...]
        at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:63)
        at java.base/java.lang.Thread.run(Thread.java:1583)
2024-07-02 16:47:06,319 [http-nio-1115-exec-7 - http://localhost:1115/xwiki/rest/liveData/sources/liveTable/entries?timestamp=1719928025842&namespace=wiki%3Axwiki&sourceParams.resultPage=XWiki.XWikiUserMembershipSheet&sourceParams.%24doc=XWiki.ILIE&sourceParams.translationPrefix=user.profile.groups.table.&sourceParams.queryFilters=currentlanguage%2Chidden&properties=group&offset=0&limit=15&sort=group&descending=] ERROR o.h.h.i.a.ErrorTracker         -  left-hand operand of a binary operator was null
2024-07-02 16:47:06,334 [http-nio-1115-exec-7 - http://localhost:1115/xwiki/rest/liveData/sources/liveTable/entries?timestamp=1719928025842&namespace=wiki%3Axwiki&sourceParams.resultPage=XWiki.XWikiUserMembershipSheet&sourceParams.%24doc=XWiki.ILIE&sourceParams.translationPrefix=user.profile.groups.table.&sourceParams.queryFilters=currentlanguage%2Chidden&properties=group&offset=0&limit=15&sort=group&descending=] ERROR o.h.h.i.a.ErrorTracker         -  left-hand operand of a binary operator was null
antlr.SemanticException: left-hand operand of a binary operator was null
        at org.hibernate.hql.internal.ast.tree.BinaryLogicOperatorNode.initialize(BinaryLogicOperatorNode.java:41)
        at org.hibernate.hql.internal.ast.HqlSqlWalker.prepareLogicOperator(HqlSqlWalker.java:1489)
   [..]
        at org.apache.tomcat.util.threads.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:659)
        at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:63)
        at java.base/java.lang.Thread.run(Thread.java:1583)
2024-07-02 16:47:06,369 [http-nio-1115-exec-7 - http://localhost:1115/xwiki/rest/liveData/sources/liveTable/entries?timestamp=1719928025842&namespace=wiki%3Axwiki&sourceParams.resultPage=XWiki.XWikiUserMembershipSheet&sourceParams.%24doc=XWiki.ILIE&sourceParams.translationPrefix=user.profile.groups.table.&sourceParams.queryFilters=currentlanguage%2Chidden&properties=group&offset=0&limit=15&sort=group&descending=] ERROR o.a.v.d.null                   - Exception in macro #gridresult_buildTagCloudJSON called at xwiki:XWiki.LiveTableResultsMacros[line 508, column 1]
2024-07-02 16:47:06,369 [http-nio-1115-exec-7 - http://localhost:1115/xwiki/rest/liveData/sources/liveTable/entries?timestamp=1719928025842&namespace=wiki%3Axwiki&sourceParams.resultPage=XWiki.XWikiUserMembershipSheet&sourceParams.%24doc=XWiki.ILIE&sourceParams.translationPrefix=user.profile.groups.table.&sourceParams.queryFilters=currentlanguage%2Chidden&properties=group&offset=0&limit=15&sort=group&descending=] ERROR o.a.v.d.null                   - Exception in macro #gridresultwithfilter_buildJSON called at xwiki:XWiki.LiveTableResultsMacros[line 532, column 1]
2024-07-02 16:47:06,370 [http-nio-1115-exec-7 - http://localhost:1115/xwiki/rest/liveData/sources/liveTable/entries?timestamp=1719928025842&namespace=wiki%3Axwiki&sourceParams.resultPage=XWiki.XWikiUserMembershipSheet&sourceParams.%24doc=XWiki.ILIE&sourceParams.translationPrefix=user.profile.groups.table.&sourceParams.queryFilters=currentlanguage%2Chidden&properties=group&offset=0&limit=15&sort=group&descending=] ERROR o.a.v.d.null                   - Exception in macro #gridresultwithfilter called at xwiki:XWiki.LiveTableResults[line 26, column 1]
```

Full stacktrace available at: https://up1.xwikisas.com/#KGlM7xfp2GaB212eXCX74A

**Comment by ilieandriuta** [comment]
Just reproduced the issue also on XWiki 16.4.1 when tried to install Collabora Connector Application (Pro) 1.3.4 (Chrome 126/ PostgreSQL 16/ Tomcat 9.0.90).

**Comment by trrenty** [comment]
I discovered a pretty consistent way to reproduce it even on Jetty + Hsqldb XWiki version 16.4.0 Java 17 Fedora 39
Could not reproduce on 14.10.

1. Have an instance with one subwiki
2. Install any extension on main wiki
3. Install the same extension on the subwiki
4. Install the said extension on farm from the main wiki

The error will pop in the logs.

I tested with Active directory, OnlyOffice, Roadmap Application from contrib, jwplayer macro.

**Comment by trrenty** [comment]
Will close this as it is not a licensing issue

**Comment by ilieandriuta** [comment]
Issue reported on product side on [https://jira.xwiki.org/browse/XWIKI-22375]( https://jira.xwiki.org/browse/XWIKI-22375) and fixed in [https://jira.xwiki.org/browse/XWIKI-22328](https://jira.xwiki.org/browse/XWIKI-22328), Fix Versions: XWiki 16.6.0-rc-1, 16.4.2 and 15.10.12.
